src/utils/settingsHandler.js

```diff
@@ -0,0 +1,137 @@
+    "setting_required": true,
+    "setting_value": null
+  },
+  {
+    "setting_name": "MongoDB URI",
+    "setting_type": "String",
+    "setting_key": "mongoDbUri",
+    "setting_required": true,
+    "setting_value": null
+  },
+  {
+    "setting_name": "Facebook Credentials",
+    "setting_type": "Object",
+    "setting_key": "facebookAuthProvider",
+    "setting_required": false,
+    "setting_value": [
+      {
+        "setting_name": "Facebook Application ID",
+        "setting_type": "String",
+        "setting_key": "appID",
+        "setting_value": null,
+        "setting_required": true
+      },
+      {
+        "setting_name": "Facebook Application Secret",
+        "setting_type": "String",
+        "setting_key": "appSecret",
+        "setting_value": null,
+        "setting_required": true
+      },
+      {
+        "setting_name": "Login Success Callback",
+        "setting_type": "String",
+        "setting_key": "successCallback",
+        "setting_required": false,
+        "setting_value": null
+      }
+    ]
+  },
+  {
+    "setting_name": "Twitter Credentials",
+    "setting_type": "Array",
+    "setting_key": "twitterAuthProvider",
+    "setting_required": false,
+    "setting_value": [
+      {
+        "setting_name": "Twitter Consumer Key",
+        "setting_type": "String",
+        "setting_key": "key",
+        "setting_value": null,
+        "setting_required": true
+      },
+      {
+        "setting_name": "Twitter Consumer Secret",
+        "setting_type": "String",
+        "setting_key": "secret",
+        "setting_value": null,
+        "setting_required": true
+      }
+    ]
+  },
+  {
+    "setting_name": "Github Credentials",
+    "setting_type": "Array",
+    "setting_key": "githubAuthProvider",
+    "setting_required": false,
+    "setting_value": [
+      {
+        "setting_name": "Github Client ID",
+        "setting_type": "String",
+        "setting_key": "clientID",
+        "setting_value": null,
+        "setting_required": true
+      },
+      {
+        "setting_name": "Github Client Secret",
+        "setting_type": "String",
+        "setting_key": "clientSecret",
+        "setting_value": null,
+        "setting_required": true
+      }
+    ]
+  },
+  {
+    "setting_name": "Google Credentials",
+    "setting_type": "Array",
+    "setting_key": "googleAuthProvider",
+    "setting_required": false,
+    "setting_value": [
+      {
+        "setting_name": "Google Client ID",
+        "setting_type": "String",
+        "setting_key": "clientID",
+        "setting_value": null,
+        "setting_required": true
+      },
+      {
+        "setting_name": "Google Client Secret",
+        "setting_type": "String",
+        "setting_key": "clientSecret",
+        "setting_value": null,
+        "setting_required": true
+      }
+    ]
+  }
+];
+
+export const parseSettings = (data) => {
+  return data.reduce((acc, item) => {
+    if (item["setting_type"] == "Object") {
```

Comment:
I think this was missed it should be `if (item["setting_type"] == "Array")`

Reply:
No, we made the change to object so as to reserve the key word for dropdown fields in forms.